lib/gui/app/components/svg-icon.js

```diff
@@ -59,31 +83,52 @@ class SVGIcon extends react.Component {
       // eslint-disable-next-line no-underscore-dangle
       : global.__dirname
 
-    // This means the path to the icon should be
-    // relative to *this directory*.
-    // TODO: There might be a way to compute the path
-    // relatively to the `index.html`.
-    const imagePath = path.join(baseDirectory, 'assets', this.props.path)
+    let svgData = ''
+
+    _.find(this.props.contents, (content) => {
+      const attempt = tryParseSVGContents(content)
 
-    let contents = ''
+      if (attempt) {
+        svgData = attempt
+        return true
+      }
+
+      return false
+    })
 
-    if (_.startsWith(this.props.path, '<')) {
-      contents = this.props.path
-    } else {
-      contents = fs.readFileSync(imagePath, {
-        encoding: 'utf8'
+    if (!svgData) {
+      _.find(this.props.paths, (relativePath) => {
+        // This means the path to the icon should be
+        // relative to *this directory*.
+        // TODO: There might be a way to compute the path
+        // relatively to the `index.html`.
+        const imagePath = path.join(baseDirectory, 'assets', relativePath)
+
+        const contents = _.attempt(() => {
+          return fs.readFileSync(imagePath, {
+            encoding: 'utf8'
+          })
+        })
+
+        if (_.isError(contents)) {
+          analytics.logException(contents)
+          return false
+        }
+
+        const parsed = _.attempt(tryParseSVGContents, contents)
+
+        if (parsed) {
+          svgData = parsed
+          return true
+        }
+
+        return false
       })
     }
 
     const width = this.props.width || DEFAULT_SIZE
     const height = this.props.height || DEFAULT_SIZE
 
-    const doc = domParser.parseFromString(contents, 'image/svg+xml')
-    const parserError = doc.querySelector('parsererror')
-    const svg = doc.querySelector('svg')
-    const svgXml = svg && _.isNil(parserError) ? svg.outerHTML : ''
-    const svgData = `data:image/svg+xml,${encodeURIComponent(svgXml)}`
-
     return react.createElement('img', {
```

Comment:
What's the behaviour if neither `contents` or `paths` resolves to an actual SVG, i.e. `svgData` is still set to `''`?